actions/internal/plugins/frontend-e2e-against-stack/action.yml

```diff
@@ -0,0 +1,296 @@
+name: Run e2e tests
+description: Run e2e tests against specific stack and environment
+inputs:
+  plugin-directory:
+    description: Directory of the plugin, if not in the root of the repository. If provided, package-manager must also be provided.
+    required: false
+    default: .
+
+  package-manager:
+    description: The package manager to use.
+    required: false
+    default: ""
+
+  npm-registry-auth:
+    description: |
+      Whether to authenticate to the npm registry in Google Artifact Registry.
+      If true, the root of the plugin repository must contain a `.npmrc` file.
+    required: false
+    default: "false"
+
+  stack_slug:
+    description: "Name of the stack where you want to run the tests"
+    required: true
+
+  env:
+    description: "Region of the stack where you want to run the tests"
+    required: true
+
+  other_plugins:
+    description: "List of other plugins that you want to enable separated by comma"
+    required: false
+
+  datasource_ids:
+    description: "List of data sources that you want to enable separated by comma"
+    required: false
+
+  upload_report_path:
+    description: "Name of the artifact where you want to store the test report"
+    required: false
+    default: "playwright-report"
+
+  upload_videos_path:
+    description: "Name of the artifact where you want to store the test videos"
+    required: false
+    default: "playwright-videos"
+
+  plugin-secrets:
+    description: "A JSON string containing key-value pairs of specific plugin secrets necessary to run the tests."
+    required: false
+
+  grafana-ini-path:
+    description: "Path to a custom grafana.ini file to configure the Grafana instance. Path should be relative to the plugin directory."
+    required: false
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        persist-credentials: false
+
+    - name: Setup Node.js environment
+      uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+      with:
+        node-version: "20"
+        cache: "yarn"
```

Comment:
These two may need to be configurable but we can keep them as-is for now until it's needed